Add NullAuthenticator to jupyterhub

manics opened the pull request:
Closes https://github.com/jupyterhub/jupyterhub/issues/3609

When JupyterHub is released we should archive https://github.com/jupyterhub/nullauthenticator

manics commented on jupyterhub/nullauthenticator.py:
```diff
@@ -0,0 +1,30 @@
+    login_service = 'null'
+
+    def get_handlers(self, app):
+        return [('/nologin', NullLoginHandler)]
```
What's the purpose of this `/nologin` handler? The original https://github.com/jupyterhub/nullauthenticator/blob/dedb18412a06da9e309c1b92c05dd305ec817d03/nullauthenticator.py#L18 uses the `web` module which isn't included in the imports, so `NullLoginHandler.get()` will return a 500 server error instead of the intended 403.

I could imagine it being used as the `login_url()` https://github.com/jupyterhub/jupyterhub/blob/b029d983f99878b365c28a6e8b606cfeb0e8d3d7/jupyterhub/auth.py#L673-L688
but that would require overriding it.

minrk replied:
I think this is a leftover from Authenticators maybe *needing* to register a handler to be redirected in earlier versions of JupyterHUb.

I think `return []` is appropriate here. The Handler itself is not needed anymore.

*alternately*, this could be `/login` and forcefully override the default login page.

The result should ultimately be the same, though, so the simplest option of an empty list and no custom handler is probably correct.

manics commented:
Thanks for the explanation. Since `Nullauthenticator` is reduced to a single small class I've moved it into auth.py instead of a separate file, and I've added a test.
